source/logic/toasts.js

```diff
@@ -11,6 +11,17 @@ function setDB(database) {
 	db = database;
 }
 
+/** *Get the ids of the rewarded Recipients on the sender's last 5 Toasts*
+ * @param {string} senderId
+ * @param {string} companyId
+ */
+async function findStaleToasteeIds(senderId, companyId) {
+	const lastFiveToasts = await db.models.Toast.findAll({ where: { senderId, companyId }, include: db.models.Toast.Recipients, order: [["createdAt", "DESC"]], limit: 5 });
+	return lastFiveToasts.reduce((list, toast) => {
+		return list.concat(toast.Recipients.filter(reciept => reciept.isRewarded).map(reciept => reciept.recipientId));
+	}, []);
```

Comment:
The IDs must be unique, right? This is probably a place to use a set, or to force the entries to be unique.

A weird way of doing that, which preserves the Array typed output:
```suggestion
	return Array.from(new Set(lastFiveToasts.reduce((list, toast) => {
		return list.concat(toast.Recipients.filter(reciept => reciept.isRewarded).map(reciept => reciept.recipientId));
	}, [])));
```

A way of doing that which integrates the Set into the `reduce`:
```suggestion
	return Array.from(lastFiveToasts.reduce((set, toast) => {
		toast.Recipients.filter(reciept => reciept.isRewarded).map(reciept => reciept.recipientId).foreach(id => set.add(id));
		return set;
	}, new Set()));
```

Reply:
Duplicated ids is intended, that way repeatedly toasting to the same hunter has accumulating staleness. I considered changing the return to `Record<userId, repeats>`, but felt the follow-up changes were out of scope.

Reply:
I suppose this is a case where the new logic layer obfuscates intention a bit. I *do* think that something like you're suggesting would add some clarity (tracking repeats is intentional) without losing potential flexibility with this function (fetching stale toastees may be useful outside the original context), but I think we can get 90% of the way there with a more verbose JSDoc (either in the initial comment or with an @returns to describe the output directly).

Reply:
The repeats map is more performant (less array searching) and concise as well, but even though I like it I'm waiting until the polish pass in case we find other functionality that would bundle well with it. Will update documentation for now.